File: backend/src/main/java/com/falconenergy/service/LoadingOrderService.java
package com.falconenergy.service;

import com.falconenergy.dto.LoadingOrderRequest;
import com.falconenergy.dto.LoadingOrderResponse;
import java.util.List;

public interface LoadingOrderService {
    LoadingOrderResponse createLoadingOrder(LoadingOrderRequest request);
    LoadingOrderResponse updateLoadingOrder(Long id, LoadingOrderRequest request);
    LoadingOrderResponse getLoadingOrderById(Long id);
    LoadingOrderResponse getLoadingOrderByOrderId(Long orderId);
    List<LoadingOrderResponse> getAllLoadingOrders();
    LoadingOrderResponse approveLoadingOrder(Long id);
    LoadingOrderResponse cancelLoadingOrder(Long id);
    LoadingOrderResponse startLoadingActivity(Long id, Long activityId, String bayNumber, String pumpNumber);
    LoadingOrderResponse completeLoadingActivity(Long id, Long activityId);
}

File: src/services/loadingOrderService.js
import { api } from "./api";

export async function createLoadingOrder(payload) {
  return api.post("/loading-orders", payload).then((r) => r.data);
}

export async function updateLoadingOrder(id, payload) {
  return api.put(`/loading-orders/${id}`, payload).then((r) => r.data);
}

export async function getLoadingOrderById(id) {
  return api.get(`/loading-orders/${id}`).then((r) => r.data);
}

export async function getLoadingOrderByOrderId(orderId) {
  return api.get(`/loading-orders/order/${orderId}`).then((r) => r.data);
}

export async function listLoadingOrders() {
  return api.get("/loading-orders").then((r) => r.data);
}

export async function approveLoadingOrder(id) {
  return api.post(`/loading-orders/${id}/approve`).then((r) => r.data);
}

export async function cancelLoadingOrder(id) {
  return api.post(`/loading-orders/${id}/cancel`).then((r) => r.data);
}

export async function startLoadingActivity(id, activityId, params = {}) {
  return api.post(`/loading-orders/${id}/activities/${activityId}/start`, null, { params }).then((r) => r.data);
}

export async function completeLoadingActivity(id, activityId) {
  return api.post(`/loading-orders/${id}/activities/${activityId}/complete`).then((r) => r.data);
}
